events/events_test.go

```diff
@@ -0,0 +1,151 @@
+package events_test
+
+import (
+	"encoding/hex"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/nsqio/go-diskqueue"
+	"go.vocdoni.io/dvote/crypto/ethereum"
+	"go.vocdoni.io/dvote/events"
+	"go.vocdoni.io/dvote/log"
+	"go.vocdoni.io/dvote/types"
+	"go.vocdoni.io/dvote/util"
+	"go.vocdoni.io/dvote/vochain"
+	"go.vocdoni.io/proto/build/go/models"
+	"google.golang.org/protobuf/proto"
+)
+
+// WARNING: This tests do not provide any vochain tx guarantee
+
+// TestCollectorOnProcessResults tests the collector with the
+// OnProcessResults event originated at the scrutinizer
+func TestCollectorOnProcessResults(t *testing.T) {
+	l := func(lvl diskqueue.LogLevel, f string, args ...interface{}) {}
+	// create dispatcher
+	signer := new(ethereum.SignKeys)
+	if err := signer.Generate(); err != nil {
+		t.Fatalf("cannot generate ethereum signer: %s", err)
+	}
+	dispatcherMock := &DispatcherMock{}
+	dispatcherMock.Dispatcher = *events.NewDispatcher(
+		signer,
+		nil,
+		t.TempDir(),
+		5000000,
+		1,
+		5000000,
+		100,
+		10000,
+		l,
+	)
+	// start dispatcher
+	dispatcherMock.StartDispatcherMock()
+
+	// create OnProcessResults event data
+	eventData := &models.ProcessResult{
+		Votes:     make([]*models.QuestionResult, 1),
+		ProcessId: Hex2byte(nil, "e9d5e8d791f51179e218c606f83f5967ab272292a6dbda887853d81f7a1d5105"),
+		EntityId:  Hex2byte(nil, "2b7222146a805bba0dbb61869c4b3a03209dffba"),
+	}
+	eventData.Votes[0] = &models.QuestionResult{
+		Question: [][]byte{{1}, {1}, {1}},
+	}
+	dispatcherMock.Collect(eventData, events.ScrutinizerOnprocessresult)
+	time.Sleep(time.Millisecond * 500)
+}
+
+func Hex2byte(tb testing.TB, s string) []byte {
+	b, err := hex.DecodeString(util.TrimHex(s))
+	if err != nil {
+		if tb == nil {
+			panic(err)
+		}
+		tb.Fatal(err)
+	}
+	return b
+}
+
+// EventWorker is responsible for performing a unit of work given an event
+type EventWorkerMock struct {
+	Storage diskqueue.Interface
```

Comment:
call this DiskQueue maybe? A few times I misunderstood this field to be a database